### DISCUSSION TITLE
Campaigns

### DISCUSSION THREAD

**Discussion by RetroGamerFan** [CorsixTH/CorsixTH#2074] [Research Room]
A few nice to have features...

1) Submit campaigns for others to download WITHIN the game MENU.

2) Optional screenshot for each level also viewable from within the game.

Anyone remember the old Micromachines game where you could easily create your own levels and enjoy them with friends?

3) Make the MAP editor even easier:

    a) Right click to rotate item like a wall or window.

    b) Undo-redo feature.

    c) Better zoom that doesn't cut-off corners of the map.

4) The "level" file options should be set in the Map designer also IMO - allow defaults based on the original game levels 1-to-12.

5) Secret levels based on [CONDITIONS HERE LIKE RATS KILLED].

etc...

**Comment by TheCycoONE** [comment]
Lot of feature requests there. Level edit in the map editor is uncontroversial.

What does better zoom mean - there already is a menu option to remove the map cutoff but the original game did not expect you to see the corners and they're frequently full of garbage.

The rest seems uncontroversial.

**Comment by RetroGamerFan** [comment]
The zoom problem is when you play a level - not the map editor (my bad). However, I don't see any menu option to remove the map cut-off (v0.65.1). You can actually edit the original levels in the map editor, there's not that much garbage. There's also garbage behind walls in some levels too - almost as if only the visible parts of levels were fixed before releasing the game originally.

The map-cut-off is a bug or limitation of the original game IMO and if you zoom in, you can scroll into the corners up-to maybe a 9x9 square of tiles which remain "hidden".

**Reply by TheCycoONE** [reply]
Enable debug options, we hid it there. It's not a bug, it's something we re-implemented on purpose. It also hides the spawning/despawning of patients.

**Reply by RetroGamerFan** [reply]
Solved - thanks for your help.

For anyone else reading...

...In "%APPDATA%\CorsixTH\config.txt", set the option "debug = true", and now when you play a level the menu at the top has a "Debug" drop-down containing many items including one called "Limit Camera".

**Comment by lewri** [comment]
For 1 see #1923 . A lot of concept and groundwork is still needed for this, but the idea is active (licences for that kind of content have currently been the hot issue here at the moment).

For 2 do you mean specifically custom levels? If so, a nice to have and would tie into 1923 again.

3 are nice things to have. It's just the map editor has been unloved for a while. 4 has also been discussed on irc/matrix/discord.

5 I believe cyco has been privately tinkering with. Over the past few months. But things have been busy for all since summer :(

**Reply by RetroGamerFan** [reply]
1. I remember reading about how competitions for kids submitting artwork basically have a clause stating the competition owners own the copy-wrongs and can do whatever they like with the works. Just make anything uploaded public domain IMO. Why complicate what should be a fun feature for others to enjoy?
2. Yes, associate a screenshot for each level of a campaign.
3. I had to use an older Map Editor for additional plots for one of my creations - it's come a long way already. Some TLC would go a long way IMO.
4. OK.
5. I can help debug and test any new features.
